pkg/utils/secrets/marshaler.go

```diff
@@ -0,0 +1,87 @@
+package secrets
+
+import (
+	"encoding/base32"
+	"fmt"
+	"strings"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+)
+
+const annotationPrefix = "flyte.secrets"
+const PodLabel = "inject-flyte-secrets"
+const PodLabelValue = "true"
+
+// Copied from:
+// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L36
+const totalAnnotationSizeLimitB int = 256 * (1 << 10) // 256 kB
+const valueSeparator = "/"
+const groupSeparator = "."
+
+var encoding = base32.StdEncoding.WithPadding(base32.NoPadding)
+
+func encodeSecretGroup(group string) string {
+	res := strings.ToLower(encoding.EncodeToString([]byte(group)))
+	return strings.TrimSuffix(res, "=")
+}
+
+func decodeSecretGroup(encoded string) (string, error) {
+	decodedRaw, err := encoding.DecodeString(strings.ToUpper(encoded))
+	if err != nil {
+		return encoded, err
+	}
+
+	return string(decodedRaw), nil
+}
+
+func MarshalSecretsToMapStrings(secrets []*core.Secret) (map[string]string, error) {
+	res := make(map[string]string, len(secrets))
```

Comment:
Why? why not keep it protobuf?

Reply:
what if you just serializeit to protobuf string and then retrieved it from the string representation?

Reply:
yeah... moved to that in latest version

Reply:
I dont follow, why is this code still here then?

Reply:
I guess the question is you are keeping every secret still a separate thing in the annotations is it? instead of just making it into one key/value?